### PR TITLE
Port ControlPart related controls to the new IPC serialization format

### PR DESCRIPTION
**Pull request by gavin-apple**
#### f074fa71d6fdf7f1bbfc1f23756d2a26b3ad4e95
<pre>
Port ControlPart related controls to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=261819">https://bugs.webkit.org/show_bug.cgi?id=261819</a>
<a href="https://rdar.apple.com/115781032">rdar://115781032</a>

Reviewed by NOBODY (OOPS!).

(relanding)

This change ports ControlPart and it&apos;s related appearances to the new IPC
serialization format. In order to do this, we&apos;ve had to extract the
ControlPartAppearance out of the ControlPart class itself. ControlPart now
acts as a wrapper around the the ControlPartAppearance, the ControlFactory,
and the PlatformControl. It&apos;s the responsibility of the ControlPartAppearance
to perform appearance specific operations on behald of the ControlPart. This
includes instantiating the appearance specific PlatformControl.

This change also adds the all of the new ControlPartAppearance to the
.serialization.in files, allowing them to be serialized within a
ControlPart. This includes:
    - ApplePayButtonAppearance,
    - ButtonAppearance,
    - ColorWellAppearance,
    - DefaultButtonAppearance,
    - ImageControlsButtonAppearance,
    - InnerSpinButtonAppearance,
    - MenuListButtonAppearance,
    - MenuListAppearance,
    - MeterAppearance,
    - ProgressBarAppearance,
    - PushButtonAppearance,
    - SearchFieldCancelButtonAppearance,
    - SearchFieldAppearance,
    - SearchFieldResultsButtonAppearance,
    - SearchFieldResultsDecorationAppearance,
    - SliderThumbHorizontalAppearance,
    - SliderThumbVerticalAppearance,
    - SliderTrackHorizontalAppearance,
    - SliderTrackVerticalAppearance,
    - SquareButtonAppearance,
    - ListBoxAppearance,
    - TextAreaAppearance,
    - TextFieldAppearance,
    - CheckboxAppearance,
    - RadioAppearance

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm:
(WebCore::ApplePayButtonCocoa::ApplePayButtonCocoa):
(WebCore::ApplePayButtonCocoa::draw):
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.h:
* Source/WebCore/platform/graphics/cocoa/controls/ControlFactoryCocoa.mm:
(WebCore::ControlFactoryCocoa::createPlatformApplePayButton):
* Source/WebCore/platform/graphics/controls/ApplePayButtonAppearance.cpp: Renamed from Source/WebCore/platform/graphics/controls/ApplePayButtonPart.cpp.
(WebCore::ApplePayButtonAppearance::ApplePayButtonAppearance):
(WebCore::ApplePayButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ApplePayButtonAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ApplePayButtonPart.h.
(WebCore::ApplePayButtonAppearance::buttonType const):
(WebCore::ApplePayButtonAppearance::setButtonType):
(WebCore::ApplePayButtonAppearance::buttonStyle const):
(WebCore::ApplePayButtonAppearance::setButtonStyle):
(WebCore::ApplePayButtonAppearance::locale const):
(WebCore::ApplePayButtonAppearance::setLocale):
* Source/WebCore/platform/graphics/controls/ButtonAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ButtonPart.h.
(WebCore::ButtonAppearanceBase::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ColorWellAppearance.h: Copied from Source/WebCore/platform/graphics/controls/ColorWellPart.h.
(WebCore::ColorWellAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ControlAppearance.h: Added.
* Source/WebCore/platform/graphics/controls/ControlFactory.h:
* Source/WebCore/platform/graphics/controls/ControlPart.cpp:
(WebCore::ControlPart::create):
(WebCore::ControlPart::ControlPart):
(WebCore::ControlPart::createPlatformControl):
(WebCore::ControlPart::appearance const):
(WebCore::ControlPart::type const):
* Source/WebCore/platform/graphics/controls/ControlPart.h:
(WebCore::ControlPart::get const):
(WebCore::ControlPart::type const): Deleted.
* Source/WebCore/platform/graphics/controls/ImageControlsButtonAppearance.h: Copied from Source/WebCore/platform/graphics/controls/ImageControlsButtonPart.h.
(WebCore::ImageControlsButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/InnerSpinButtonAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/MenuListButtonPart.h.
(WebCore::InnerSpinButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/MenuListAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/MenuListPart.h.
(WebCore::MenuListAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/MenuListButtonAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/InnerSpinButtonPart.h.
(WebCore::MenuListButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/MeterAppearance.cpp: Renamed from Source/WebCore/platform/graphics/controls/MeterPart.cpp.
(WebCore::MeterAppearance::MeterAppearance):
(WebCore::MeterAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/MeterAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/MeterPart.h.
(WebCore::MeterAppearance::gaugeRegion const):
(WebCore::MeterAppearance::setGaugeRegion):
(WebCore::MeterAppearance::value const):
(WebCore::MeterAppearance::setValue):
(WebCore::MeterAppearance::minimum const):
(WebCore::MeterAppearance::setMinimum):
(WebCore::MeterAppearance::maximum const):
(WebCore::MeterAppearance::setMaximum):
* Source/WebCore/platform/graphics/controls/PlatformControl.h:
(WebCore::PlatformControl::owningPart const):
* Source/WebCore/platform/graphics/controls/ProgressBarAppearance.cpp: Renamed from Source/WebCore/platform/graphics/controls/ProgressBarPart.cpp.
(WebCore::ProgressBarAppearance::ProgressBarAppearance):
(WebCore::ProgressBarAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ProgressBarAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ProgressBarPart.h.
(WebCore::ProgressBarAppearance::position const):
(WebCore::ProgressBarAppearance::setPosition):
(WebCore::ProgressBarAppearance::animationStartTime const):
(WebCore::ProgressBarAppearance::setAnimationStartTime):
* Source/WebCore/platform/graphics/controls/SearchFieldAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/TextFieldPart.h.
(WebCore::SearchFieldAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/SearchFieldCancelButtonAppearance.h: Copied from Source/WebCore/platform/graphics/controls/SearchFieldPart.h.
(WebCore::SearchFieldCancelButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/SearchFieldResultsAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ImageControlsButtonPart.h.
(WebCore::SearchFieldResultsPart::createPlatformControl):
* Source/WebCore/platform/graphics/controls/SliderThumbAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SearchFieldResultsPart.h.
(WebCore::SliderThumbAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/SliderThumbPart.h: Removed.
* Source/WebCore/platform/graphics/controls/SliderTrackAppearance.cpp: Renamed from Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp.
(WebCore::SliderTrackAppearance::SliderTrackAppearance):
(WebCore::SliderTrackAppearance::drawTicks const):
(WebCore::SliderTrackHorizontalAppearance::drawTicks const):
(WebCore::SliderTrackVerticalAppearance::drawTicks const):
(WebCore::SliderTrackAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/SliderTrackAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SliderTrackPart.h.
(WebCore::SliderTrackAppearance::thumbSize const):
(WebCore::SliderTrackAppearance::setThumbSize):
(WebCore::SliderTrackAppearance::trackBounds const):
(WebCore::SliderTrackAppearance::setTrackBounds):
(WebCore::SliderTrackAppearance::tickRatios const):
(WebCore::SliderTrackAppearance::setTickRatios):
* Source/WebCore/platform/graphics/controls/SwitchThumbAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SwitchThumbPart.h.
* Source/WebCore/platform/graphics/controls/SwitchTrackAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SwitchTrackPart.h.
* Source/WebCore/platform/graphics/controls/TextAreaAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SearchFieldCancelButtonPart.h.
(WebCore::TextAreaAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/TextAreaPart.h: Removed.
* Source/WebCore/platform/graphics/controls/TextFieldAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/SearchFieldPart.h.
(WebCore::TextFieldAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ToggleButtonAppearance.h: Renamed from Source/WebCore/platform/graphics/controls/ColorWellPart.h.
(WebCore::ToggleButtonAppearance::createPlatformControl):
* Source/WebCore/platform/graphics/controls/ToggleButtonPart.h: Removed.
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.h:
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.mm:
(WebCore::ControlFactoryIOS::createPlatformButton):
(WebCore::ControlFactoryIOS::createPlatformColorWell):
(WebCore::ControlFactoryIOS::createPlatformInnerSpinButton):
(WebCore::ControlFactoryIOS::createPlatformMenuList):
(WebCore::ControlFactoryIOS::createPlatformMenuListButton):
(WebCore::ControlFactoryIOS::createPlatformMeter):
(WebCore::ControlFactoryIOS::createPlatformProgressBar):
(WebCore::ControlFactoryIOS::createPlatformSearchField):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldCancelButton):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldResults):
(WebCore::ControlFactoryIOS::createPlatformSliderThumb):
(WebCore::ControlFactoryIOS::createPlatformSliderTrack):
(WebCore::ControlFactoryIOS::createPlatformSwitchThumb):
(WebCore::ControlFactoryIOS::createPlatformSwitchTrack):
(WebCore::ControlFactoryIOS::createPlatformTextArea):
(WebCore::ControlFactoryIOS::createPlatformTextField):
(WebCore::ControlFactoryIOS::createPlatformToggleButton):
* Source/WebCore/platform/graphics/mac/controls/ButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/ButtonMac.mm:
(WebCore::ButtonMac::ButtonMac):
* Source/WebCore/platform/graphics/mac/controls/ColorWellMac.h:
* Source/WebCore/platform/graphics/mac/controls/ColorWellMac.mm:
(WebCore::ColorWellMac::ColorWellMac):
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.mm:
* Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm:
(WebCore::ImageControlsButtonMac::ImageControlsButtonMac):
* Source/WebCore/platform/graphics/mac/controls/InnerSpinButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/InnerSpinButtonMac.mm:
(WebCore::InnerSpinButtonMac::InnerSpinButtonMac):
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::MenuListButtonMac::MenuListButtonMac):
* Source/WebCore/platform/graphics/mac/controls/MenuListMac.h:
* Source/WebCore/platform/graphics/mac/controls/MenuListMac.mm:
(WebCore::MenuListMac::MenuListMac):
* Source/WebCore/platform/graphics/mac/controls/MeterMac.h:
* Source/WebCore/platform/graphics/mac/controls/MeterMac.mm:
(WebCore::MeterMac::MeterMac):
(WebCore::MeterMac::updateCellStates):
* Source/WebCore/platform/graphics/mac/controls/ProgressBarMac.h:
* Source/WebCore/platform/graphics/mac/controls/ProgressBarMac.mm:
(WebCore::ProgressBarMac::ProgressBarMac):
(WebCore::ProgressBarMac::draw):
* Source/WebCore/platform/graphics/mac/controls/SearchFieldCancelButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldCancelButtonMac.mm:
(WebCore::SearchFieldCancelButtonMac::SearchFieldCancelButtonMac):
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.h:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.mm:
(WebCore::SearchFieldMac::SearchFieldMac):
* Source/WebCore/platform/graphics/mac/controls/SearchFieldResultsMac.h:
* Source/WebCore/platform/graphics/mac/controls/SearchFieldResultsMac.mm:
(WebCore::SearchFieldResultsMac::SearchFieldResultsMac):
* Source/WebCore/platform/graphics/mac/controls/SliderThumbMac.h:
* Source/WebCore/platform/graphics/mac/controls/SliderThumbMac.mm:
(WebCore::SliderThumbMac::SliderThumbMac):
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.h:
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.mm:
(WebCore::SliderTrackMac::SliderTrackMac):
(WebCore::SliderTrackMac::rectForBounds const):
(WebCore::SliderTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/SwitchThumbMac.h:
* Source/WebCore/platform/graphics/mac/controls/SwitchThumbMac.mm:
(WebCore::SwitchThumbMac::SwitchThumbMac):
(WebCore::SwitchThumbMac::draw):
* Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.h:
* Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.mm:
(WebCore::SwitchTrackMac::SwitchTrackMac):
(WebCore::SwitchTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/TextAreaMac.h:
* Source/WebCore/platform/graphics/mac/controls/TextAreaMac.mm:
(WebCore::TextAreaMac::TextAreaMac):
* Source/WebCore/platform/graphics/mac/controls/TextFieldMac.h:
* Source/WebCore/platform/graphics/mac/controls/TextFieldMac.mm:
(WebCore::TextFieldMac::TextFieldMac):
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.h:
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.mm:
(WebCore::ToggleButtonMac::ToggleButtonMac):
* Source/WebCore/rendering/RenderMenuList.cpp:
(RenderMenuList::clientPaddingLeft const):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
(WebCore::updateApplePayButtonAppearanceForRenderer):
(WebCore::updateMeterAppearanceForRenderer):
(WebCore::updateProgressBarAppearanceForRenderer):
(WebCore::updateSliderTrackAppearanceForRenderer):
(WebCore::updateSwitchThumbAppearanceForRenderer):
(WebCore::updateSwitchTrackAppearanceForRenderer):
(WebCore::RenderTheme::createControlPart const):
(WebCore::RenderTheme::updateControlPartForRenderer const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::isControlStyled const):
(WebCore::RenderTheme::adjustSearchFieldResultsDecorationAppearanceStyle const):
(WebCore::RenderTheme::purgeCaches):
(WebCore::RenderTheme::platformColorsDidChange):
(WebCore::updateApplePayButtonPartForRenderer): Deleted.
(WebCore::updateMeterPartForRenderer): Deleted.
(WebCore::updateProgressBarPartForRenderer): Deleted.
(WebCore::updateSliderTrackPartForRenderer): Deleted.
(WebCore::updateSwitchThumbPartForRenderer): Deleted.
(WebCore::updateSwitchTrackPartForRenderer): Deleted.
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::paintSearchFieldResultsDecorationAppearance):
(WebCore::RenderTheme::adjustSearchFieldResultsDecorationPartStyle const): Deleted.
(WebCore::RenderTheme::paintSearchFieldResultsDecorationPart): Deleted.
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
* Source/WebCore/rendering/ios/RenderThemeIOS.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::adjustSearchFieldResultsDecorationAppearanceStyle const):
(WebCore::RenderThemeIOS::paintSearchFieldResultsDecorationAppearance):
(WebCore::RenderThemeIOS::adjustSearchFieldResultsDecorationPartStyle const): Deleted.
(WebCore::RenderThemeIOS::paintSearchFieldResultsDecorationPart): Deleted.
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustSearchFieldResultsDecorationAppearanceStyle const):
(WebCore::RenderThemeMac::adjustSearchFieldResultsDecorationPartStyle const): Deleted.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;ControlPart&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;ControlPart&gt;::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
</pre>
https://github.com/WebKit/WebKit/commit/f074fa71d6fdf7f1bbfc1f23756d2a26b3ad4e95

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37462 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16346 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39725 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39988 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33384 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38773 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/19007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13505 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31794 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38027 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13785 "Found 1 new test failure: fast/forms/ios/autocapitalize-words.html (failure)") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32876 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11993 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11978 "Unexpected infrastructure issue, retrying build") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33556 "Found 21 new test failures: fast/forms/switch/click-animation-disabled.html, fast/forms/switch/click-animation-interrupted.html, fast/forms/switch/click-animation-preventdefault.html, fast/forms/switch/click-animation-redundant-checked.html, fast/forms/switch/click-animation-redundant-disabled.html, fast/forms/switch/click-animation-twice-fast.html, fast/forms/switch/click-animation-twice.html, fast/forms/switch/click-animation.html, fast/forms/switch/no-pixels-outside-the-box-vertical-lr-rtl.html, fast/forms/switch/no-pixels-outside-the-box.html ... (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41251 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33868 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/33975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37873 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12534 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/10162 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36032 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12945 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |

### CODE REVIEW

File: Source/WebCore/Headers.cmake [hunk 1]
```diff
@@ -2047,28 +2047,29 @@ set(WebCore_PRIVATE_FRAMEWORK_HEADERS
     platform/graphics/angle/ANGLEUtilities.h
     platform/graphics/angle/GraphicsContextGLANGLE.h
 
-    platform/graphics/controls/ButtonPart.h
-    platform/graphics/controls/ColorWellPart.h
+    platform/graphics/controls/ButtonAppearance.h
```
Comment (cdumez) [inline Source/WebCore/Headers.cmake:2050]: I have an alternative proposal at https://github.com/WebKit/WebKit/pull/23713 which doesn't require refactoring the code at all.

File: Source/WebKit/WebKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1679,7 +1679,7 @@
 		8644890B27B47020007A1C66 /* _WKSystemPreferences.h in Headers */ = {isa = PBXBuildFile; fileRef = 8644890A27B47020007A1C66 /* _WKSystemPreferences.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		8644890F27B5CB43007A1C66 /* _WKSystemPreferencesInternal.h in Headers */ = {isa = PBXBuildFile; fileRef = 8644890E27B5CB43007A1C66 /* _WKSystemPreferencesInternal.h */; };
 		866623F929C2248E0029405A /* WebAutocorrectionData.mm in Sources */ = {isa = PBXBuildFile; fileRef = 866623F829C2248D0029405A /* WebAutocorrectionData.mm */; };
-		86760A6B28DB30DE00D07D06 /* GeneratedSerializers.mm in Sources */ = {isa = PBXBuildFile; fileRef = 86760A6928DB305F00D07D06 /* GeneratedSerializers.mm */; };
+		86760A6B28DB30DE00D07D06 /* GeneratedSerializers.mm in Sources */ = {isa = PBXBuildFile; fileRef = 86760A6928DB305F00D07D06 /* GeneratedSerializers.mm */; settings = {COMPILER_FLAGS = "-Os"; }; };
```
Comment (annevk) [inline Source/WebKit/WebKit.xcodeproj/project.pbxproj:1682]: Is this an intentional change?

**Review by annevk — COMMENTED** [review]
I don't think I'm a good reviewer for this, though the bits of the code I'm familiar with look reasonable. I like the that the new naming more closely follows CSS.

I would have expected the commit message to explain why the previous approach was backed out and how this differs from that in order to not get backed out again.

File: Source/WebKit/WebKit.xcodeproj/project.pbxproj (hunk 2 above)
Comment (cdumez) [reply Source/WebKit/WebKit.xcodeproj/project.pbxproj:1682]: I think this part of this change is worth trying to see the effect on our binary size. However, it should be done in its own PR.

Comment (gavin-apple) [reply Source/WebKit/WebKit.xcodeproj/project.pbxproj:1682]: Yes, intentional. Yes, we should try in its own PR. Closing this one in favour of Chris's other change

**Comment by gavin-apple** [comment]
> I don't think I'm a good reviewer for this, though the bits of the code I'm familiar with look reasonable. I like the that the new naming more closely follows CSS.
> 
> I would have expected the commit message to explain why the previous approach was backed out and how this differs from that in order to not get backed out again.

Previous change was backed out due to code size increase. We believe that was due to the GeneratedSerializers.mm blowing up from the variant usage.

**Pull request closed without merging**